housekeeping/src/main/java/com/hotels/housekeeping/service/impl/FileSystemHousekeepingService.java

```diff
@@ -172,6 +191,11 @@ Path deleteParents(FileSystem fs, Path path, String pathEventId) throws IOExcept
   }
 
   private boolean isEmpty(FileSystem fs, Path path) throws IOException {
+    // TODO PD this is wrong for S3AFileSystems when you have a path that has those empty ..$folder$ files.
```

Comment:
I am working on fixing this TODO it will involve wrapping around the FileSystem that we use here. Side effect of that is that we can then mock it in the test so hopefully the issues I see will go away.